nucliadb/nucliadb/standalone/auth.py

```diff
@@ -153,12 +188,30 @@ async def authenticate(
         return auth_creds, nuclia_user
 
 
-def get_auth_backend(settings: Settings) -> AuthenticationBackend:
-    if settings.auth_policy == AuthPolicy.UPSTREAM_NAIVE:
-        return NucliaCloudAuthenticationBackend(
+class UpstreamNaiveAuthenticationBackend(
+    AuthTokenAuthenticationBackend, NucliaCloudAuthenticationBackend
+):
+    def __init__(self, settings: Settings) -> None:
+        AuthTokenAuthenticationBackend.__init__(self, settings)
+        NucliaCloudAuthenticationBackend.__init__(
+            self,
             roles_header=settings.auth_policy_roles_header,
             user_header=settings.auth_policy_user_header,
         )
+
+    async def authenticate(
+        self, request: HTTPConnection
+    ) -> Optional[Tuple[AuthCredentials, BaseUser]]:
+        token_resp = await AuthTokenAuthenticationBackend.authenticate(self, request)
```

Comment:
NIT: I think this pattern is better than the one above (inheritance) because it's more explicit.

Reply:
I think you're right. I will tweak the PR to make it use less inheritance.